### PR TITLE
FOUR-10447: Add Preview for screens

### PR DESCRIPTION
**Pull request by danloa**
## Issue & Reproduction Steps
Adds the Preview for Screens. This preview will be used in modeler.

image

## Related Tickets & Packages
- [https://processmaker.atlassian.net/browse/FOUR-10447](https://processmaker.atlassian.net/browse/FOUR-10447)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy
ci:next
ci:modeler:feature/FOUR-10447
ci:package-decision-engine:feature/FOUR-10449

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-757dfb090c.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-757dfb090c.eng.processmaker.net

File: resources/js/processes/modeler/initialLoad.js [hunk 1]
```diff
@@ -465,4 +465,17 @@ ProcessMaker.EventBus.$on(
   },
 );
 
+ProcessMaker.EventBus.$on(
+  "modeler-init",
+  (evento) => {
+    evento.registerPreview({
```
Comment (josechirivella) [inline resources/js/processes/modeler/initialLoad.js:471]:
```suggestion
  (event) => {
    event.registerPreview({
```

File: resources/js/processes/modeler/initialLoad.js [hunk 2]
```diff
@@ -465,4 +465,17 @@ ProcessMaker.EventBus.$on(
...
+      url:'/designer/screens/preview',
+      receivingParams: ['screenRef'],
+      matcher: (nodeData) => {
+        return nodeData.$type && nodeData.$type === 'bpmn:Task'
```
Comment (josechirivella) [inline resources/js/processes/modeler/initialLoad.js:475]: try to see if this is compiled
```suggestion
        return nodeData?.$type === 'bpmn:Task'
```

**Review by josechirivella — CHANGES_REQUESTED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-757dfb090c.eng.processmaker.net

File: resources/js/processes/scripts/components/ScriptPreview.vue [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+          })
+          .then(response => {
+            ProcessMaker.alert(this.$t("The script was saved."), "success");
+            if (typeof resolve === "function") {
+              resolve(response);
+            }
+          }).catch(err => {
+        if (typeof reject === "function") {
+          reject(err);
+        }
```
Comment (josechirivella) [inline resources/js/processes/scripts/components/ScriptPreview.vue:184]: I dont think this is necessary? 
```suggestion
  console.error(err)
```

File: resources/js/processes/scripts/components/ScriptPreview.vue [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+      });
+
+      if (response.status !== 200) {
+        this.preview.executing = false;
+        this.preview.failure = true;
+        this.preview.error.exception = response.status;
+        this.preview.error.message = response.response;
+      }
+    },
+    stopResizing: _.debounce(function() {
```
Comment (josechirivella) [inline resources/js/processes/scripts/components/ScriptPreview.vue:140]:
```suggestion
    stopResizing: debounce(function() {
```

File: resources/js/processes/scripts/components/ScriptPreview.vue [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+          <i v-if="preview.failure" class="fas fa-times-circle text-danger"></i>
+        </span>
+      </b-card-footer>
+    </b-card>
+  </b-container>
+</template>
+
+<script>
+import MonacoEditor from "vue-monaco";
+import _ from "lodash";
```
Comment (josechirivella) [inline resources/js/processes/scripts/components/ScriptPreview.vue:35]:
```suggestion
import { debounce } from "lodash";
```

**Review by josechirivella — CHANGES_REQUESTED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-757dfb090c.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-757dfb090c.eng.processmaker.net

**Review by josechirivella — APPROVED** [review]

**Pull request merged**